content/docs/examples/advanced-gateways/http-proxy/index.md

```diff
@@ -0,0 +1,240 @@
+
+    {{< text bash >}}
+    $ kubectl create configmap proxy-configmap -n external --from-file=squid.conf=./proxy.conf
+    {{< /text >}}
+
+1.  Deploy a container with Squid:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      name: squid
+      namespace: external
+    spec:
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: squid
+        spec:
+          volumes:
+          - name: proxy-config
+            configMap:
+              name: proxy-configmap
+          containers:
+          - name: squid
+            image: sameersbn/squid:3.5.27
+            imagePullPolicy: IfNotPresent
+            volumeMounts:
+            - name: proxy-config
+              mountPath: /etc/squid
+              readOnly: true
+    EOF
+    {{< /text >}}
+
+1.  Deploy the [sleep]({{< github_tree >}}/samples/sleep) sample in the `external` namespace to test traffic to the
+    proxy without Istio traffic control.
+
+    {{< text bash >}}
+    $ kubectl apply -n external -f @samples/sleep/sleep.yaml@
+    {{< /text >}}
+
+1.  Define an environment variable to hold the IP address of the proxy pod:
+
+    {{< text bash >}}
+    $ export PROXY_IP=$(kubectl get pod -n external -l app=squid -o jsonpath={.items..podIP})
+    {{< /text >}}
+
+1.  Define an environment variable to hold the port of your proxy. The deployment of Squid in this example uses port
+    3128.
+
+    {{< text bash >}}
+    $ export PROXY_PORT=3128
+    {{< /text >}}
+
+1.  Send a request from the `sleep` pod in the `external` namespace to an external service via the proxy:
+
+    {{< text bash >}}
+    $ kubectl exec -it $(kubectl get pod -n external -l app=sleep -o jsonpath={.items..metadata.name}) -n external -- sh -c "HTTPS_PROXY=$PROXY_IP:$PROXY_PORT curl https://en.wikipedia.org/wiki/Main_Page" | grep -o "<title>.*</title>"
+    <title>Wikipedia, the free encyclopedia</title>
+    {{< /text >}}
+
+1.  Check the access log of the proxy and see your request:
+
+    {{< text bash >}}
+    $ kubectl exec -it $(kubectl get pod -n external -l app=squid -o jsonpath={.items..metadata.name}) -n external -- tail -f /var/log/squid/access.log
+    1544160065.248    228 172.30.109.89 TCP_TUNNEL/200 87633 CONNECT en.wikipedia.org:443 - HIER_DIRECT/91.198.174.192 -
+    {{< /text >}}
+
+At this point the proxy has been deployed and tested by using `curl` to access `wikipedia.org` through the proxy, all
+without Istio. It's just plain Kubernetes setting so far, which simulates an external HTTPS proxy.
+In the following section you are going to configure traffic from Istio-enabled pods to the HTTPS proxy.
+
+## Configure traffic to external HTTPS proxy
+
+1.  Define a TCP (!) Service Entry for the HTTPS proxy. Note that despite the fact that HTTP
+    [CONNECT](https://tools.ietf.org/html/rfc7231#section-4.3.6) method is used to communicate with HTTPS proxies,
```

Comment:
that the HTTP

Reply:
Fixed in https://github.com/istio/istio.io/pull/2960/commits/2ea97e6e2b9c4682a29b7e108ae88b8b60c80822

Reply:
Passive voice issue: What is using the HTTP CONNECT method to communicate with the HTTPS proxies?

Reply:
@rcaballeromx I would like to raise the issue of the passive voice again. cc @frankbu @geeknoid 

According to https://writing.stackexchange.com/questions/16812/technical-documentation-voice-preference-active-voice-passive-voice-a, which cites an APA manual:

> The passive voice is acceptable in expository writing and when you want to focus on the object or recipient of the action rather than on the actor. For example, "The speakers were attached to either side of the chair" emphasizes the placement of speakers, not who placed them—the more appropriate focus in the Method section. "The President was shot" emphasizes the importance of the person shot. (p. 77)

Kubernetes documentation uses passive voice extensively:

Setup https://kubernetes.io/docs/setup/

> These solutions allow you to create Kubernetes clusters with only a few commands and are actively developed

> They can also be hosted on a range of Cloud IaaS providers

Concepts: https://kubernetes.io/docs/concepts/

> The master can also be replicated for availability and redundancy.

Install kubectl
https://kubernetes.io/docs/tasks/tools/install-kubectl/

> which is created automatically
>  The completion script itself is generated by kubectl
>  which is not installed by default
> so it is automatically loaded

My claim is that passive voice should be used to omit obvious or unimportant details.

Reply:
@vadimeisenbergibm This topic is indeed a challenging one but let me try to explain why, passive voice is not appropriate for this type of content.
1. As you point out, passive voice does have a role in writing. There are reasons why it exists and the APA manual cites them very well. However, the American Press Association manual pertains to journalistic content, not technical content. The oldest and best established style guide for developer facing content is the [Microsoft Style guide](https://docs.microsoft.com/en-us/style-guide/welcome/) which recommends active voice. If you are interested, I can provide you a full literature list of technical writing books which emphasize the importance of active voice in technical content. OpenStack, the Linux kernel, Google and many other projects also have technical content guidelines recommending the use of active voice.
2. It is very difficult to accurately know what is obvious or unimportant to your readers. The best way we have found to leave as few readers behind as possible, is to attempt to be as explicit, clear, specific, and accurate as possible. In this particular case, emphasizing the target and omitting the actor makes your content harder to grasp and less specific.
3. I will raise the issue of passive voice in Kubernetes with that technical writing team but please keep in mind that, similar to Istio, not every PR from every contributor gets a full review by an editor or a tech. writer. Thus, instances of passive voice might slip through. Not because they are considered correct, but because there was no chance to correct them. The [Kubernetes content best practices](https://kubernetes.io/docs/contribute/style/style-guide/#content-best-practices) make the clear recommendation to use active voice, with the sole exception its use leading to strange constructs. That is not the case in this instance.
The use of active voice in technical content is and has been a technical writing best practice for many years and across many languages. If you want more information or more resources regarding this topic, please let me know.

Reply:
I understand the general guidance but I also think that some of @vadimeisenbergibm's examples fall into a legitimate exception category (active voice adds unimportant details). In [this document](https://writing.wisc.edu/Handbook/CCS_activevoice.html) they call out exactly this case as an exception.
```
To describe a condition in which the actor is unknown or unimportant
```
Passive voice:
```
Every year, thousands of people are diagnosed as having cancer. 
```
Active voice would be adding unimportant detail to the message, e.g.:
```
Every year, doctors diagnose thousands of people as having cancer.
```
I think this is the same issue Vadim is having, at least in some of the cases.

Reply:
@rcaballeromx Thank you for the extensive answer and for the references. My point is as follows: While I agree we should strive to use active voice as much as possible, there could be cases where passive voice could be beneficial. So using passive voice should not be rejected outright, we should weigh the alternatives in each case.

My concrete proposal is: in every PR the authors will document all the cases of the passive voice and explain why they used them and which alternatives they considered. This way we will have all the passive voice cases under control, and it will be clear that passive voice was not added by omission, but intentionally by the author. Then we can discuss if each case merits using passive voice.

Reply:
@rcaballeromx Agreed, for the new PR I listed the passive voice cases in the PR's first comment.

Reply:
@rcaballeromx I agree with you, although your wording in this example is not quite correct and misses the point. Let me give it a try.

Note that even though applications use the HTTP CONNECT method to communicate with HTTPS proxies, Istio will not handle these requests as HTTP traffic. It will instead forward them as opaque TCP requests (i.e., a TCP tunnel).

Reply:
@frankbu That's why I love collaborating on content, if I may improve upon your version:
Although applications use the HTTP CONNECT method to communicate with HTTPS proxies, Istio doesn't handle these requests as HTTP traffic. Instead, Istio forwards them as opaque TCP requests, for example, as a TCP tunnel.
Unlike before, my suggestions here are mostly copy-editing to fix things like verb congruence, unclear pronoun references, and that sort of thing.

Reply:
just change "for example", to "that is" and I like it.

@vadimeisenbergibm WDYT?

Reply:
Actually I think you need to also change "doesn't" to "won't" and "forwards" to "will forward", because this will only be the case after you apply the configuration that comes right after this sentence.

Although applications use the HTTP CONNECT method to communicate with HTTPS proxies, Istio won't handle these requests as HTTP traffic. Instead, Istio will forward them as opaque TCP requests, that is, as a TCP tunnel.

Reply:
Another try:

Although applications use the HTTP CONNECT method to establish connections with HTTPS proxies, you must configure the proxy for TCP traffic, instead of HTTP, because once a connection is established the proxy simply acts as a TCP tunnel.

Reply:
Fixed in https://github.com/istio/istio.io/pull/2973/commits/96f9545aa7c16404aa79479b710daaaae8759002